### ISSUE TITLE
HCCleanupAsync deadlock on HCWebSocketConnectAsync failure.

### ISSUE THREAD

**Issue by AndriiCD** [microsoft/libHttpClient#909]
On XBOX GDK 250400 version after HCWebSocketConnectAsync failure, the HCCleanupAsync creates deadlock and cannot be finished or even terminated. Possible reason that I found in logs is :
```
[0758][I][16:01:09.365][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=11
[0758][I][16:01:09.417][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=21
[0758][I][16:01:09.418][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=41
[0758][I][16:01:09.421][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=81
[0758][I][16:01:09.525][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=321
[0758][I][16:01:09.551][HTTPCLIENT] HCHttpCallPerform [ID 2] [TID 1880l] WINHTTP_CALLBACK_STATUS_SENDREQUEST_COMPLETE
[0758][I][16:01:09.551][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=641
[0758][I][16:01:09.551][HTTPCLIENT] WinHttpConnection WinHttp callback statusCode=1281
[0758][I][16:01:09.551][HTTPCLIENT] HCHttpCallPerform [ID 2] [TID 1880l] Websocket WINHTTP_CALLBACK_STATUS_HEADERS_AVAILABLE
[0758][E][16:01:09.551][HTTPCLIENT] HCHttpCallPerform [ID 2] [TID 1880l] WinHttpWebSocketCompleteUpgrade errorcode 4317
[0758][I][16:01:09.551][HTTPCLIENT] HCHttpCallResponseSetErrorCode [ID 2]: errorCode=80004005 (800710DD)
[0758][V][16:01:09.551][HTTPCLIENT] WinHttpConnection::StartWinHttpClose, current state=0 transitioning to ConnectionState::WinHttpClosing
[0758][E][16:01:09.551][HTTPCLIENT] WinHttpCloseHandle failed with errorCode=6
[0730][I][16:01:11.418][WEBSOCKET] HC_WEBSOCKET_OBSERVER::~HC_WEBSOCKET_OBSERVER
```
WinHttpCloseHandle couldn't be cleaned up, so it provokes deadlock.

**Comment by AndriiCD** [comment]
Here is code to reproduce it, took part of it from sample app of github repo. On newest version took from your repo that works fine, but on GDK version( 250400 ) it has issue with cleaning up. Suspicious log line for me is
[0758][E][16:01:09.551][HTTPCLIENT] WinHttpCloseHandle failed with errorCode=6
`struct WebSocketContext
{
	WebSocketContext()
	{
		closeEventHandle = CreateEvent(nullptr, false, false, nullptr);
	}

	~WebSocketContext()
	{
		if (handle)
		{
			HCWebSocketCloseHandle(handle);
		}
		CloseHandle(closeEventHandle);
	}

	HCWebsocketHandle handle{nullptr};
	std::vector<char> receiveBuffer;
	std::atomic<uint32_t> messagesReceived;
	HANDLE closeEventHandle;
};

XTaskQueueHandle g_queue;
XTaskQueueRegistrationToken g_callbackToken;

int main(int /*argc*, char** /*argv*/)
{
	HCInitialize(nullptr);
	HCSettingsSetTraceLevel(HCTraceLevel::Verbose);

	XTaskQueueCreate(XTaskQueueDispatchMode::Manual, XTaskQueueDispatchMode::Manual, &g_queue);
	std::thread webSocketThread{[]()
								{
									HCTraceSetTraceToDebugger(true);
									HCSettingsSetTraceLevel(HCTraceLevel::Verbose);
									while (true)
									{
										XTaskQueueDispatch(g_queue, XTaskQueuePort::Work, 0);
										XTaskQueueDispatch(g_queue, XTaskQueuePort::Completion, 0);
									}
								}};
	std::string url =
		"wss://rtm.live.cdpred.services/v1/?user_id=2795867471933132&client_id=client_id";

	auto websocketContext = new WebSocketContext{};
	HCWebSocketCreate(
		&websocketContext->handle,
		nullptr,
		nullptr, nullptr,
		websocketContext
	);
	HCWebSocketSetBinaryMessageFragmentEventFunction(
		websocketContext->handle, nullptr
	);
	HCWebSocketSetMaxReceiveBufferSize(websocketContext->handle, 4096);
	XAsyncBlock* asyncBlock = new XAsyncBlock{};
	asyncBlock->queue = g_queue;
	asyncBlock->callback = [](XAsyncBlock* asyncBlock)
	{
		WebSocketCompletionResult result = {};
		HRESULT hr = HCGetWebSocketConnectResult(asyncBlock, &result);
		assert(SUCCEEDED(hr));

		if (SUCCEEDED(hr))
		{
			printf_s(
				"HCWebSocketConnect complete: %d, %d\n",
				result.errorCode,
				result.platformErrorCode
			);
			if (FAILED(result.errorCode))
			{
				// throw std::exception("Connect failed. Make sure a local echo server is
				// running.");
			}
		}

		delete asyncBlock;
	};

	printf_s("Calling HCWebSocketConnect...\n");
	HCWebSocketConnectAsync(url.data(), "", websocketContext->handle, asyncBlock);
	XAsyncGetStatus(asyncBlock, true);

	printf_s("Calling HCWebSocketDisconnect...\n");
	HCWebSocketDisconnect(websocketContext->handle);
	delete websocketContext;
	auto* block = new XAsyncBlock{};

	HCCleanupAsync(block);
	XAsyncGetStatus(block, true);
	webSocketThread.detach();
	return 0;
}
`
